build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/release/ReleaseToolsPlugin.java

```diff
@@ -0,0 +1,97 @@
+            .getAsFileTree()
+            .matching(new PatternSet().include("**/*.yml", "**/*.yaml"));
+
+        final Provider<ValidateYamlAgainstSchemaTask> validateChangelogsAgainstYamlTask = project.getTasks()
+            .register("validateChangelogsAgainstSchema", ValidateYamlAgainstSchemaTask.class, task -> {
+                task.setGroup("Documentation");
+                task.setDescription("Validate that the changelog YAML files comply with the changelog schema");
+                task.setInputFiles(yamlFiles);
+                task.setJsonSchema(new File(project.getRootDir(), RESOURCES + "changelog-schema.json"));
+                task.setReport(new File(project.getBuildDir(), "reports/validateYaml.txt"));
+            });
+
+        final TaskProvider<ValidateChangelogEntryTask> validateChangelogsTask = project.getTasks()
+            .register("validateChangelogs", ValidateChangelogEntryTask.class, task -> {
+                task.setGroup("Documentation");
+                task.setDescription("Validate that all changelog YAML files are well-formed");
+                task.setChangelogs(yamlFiles);
+                task.dependsOn(validateChangelogsAgainstYamlTask);
+            });
+
+        project.getTasks().register("generateReleaseNotes", GenerateReleaseNotesTask.class).configure(task -> {
+            final Version version = VersionProperties.getElasticsearchVersion();
+
+            task.setGroup("Documentation");
+            task.setDescription("Generates release notes from changelog files held in this checkout");
+            task.setChangelogs(yamlFiles);
+
+            task.setReleaseNotesIndexTemplate(projectDirectory.file(RESOURCES + "templates/release-notes-index.asciidoc"));
+            task.setReleaseNotesIndexFile(projectDirectory.file("docs/reference/release-notes.asciidoc"));
+
+            task.setReleaseNotesTemplate(projectDirectory.file(RESOURCES + "templates/release-notes.asciidoc"));
+            task.setReleaseNotesFile(
+                projectDirectory.file(String.format("docs/reference/release-notes/%d.%d.asciidoc", version.getMajor(), version.getMinor()))
+            );
+
+            task.setReleaseHighlightsTemplate(projectDirectory.file(RESOURCES + "templates/release-highlights.asciidoc"));
+            task.setReleaseHighlightsFile(projectDirectory.file("docs/reference/release-notes/highlights.asciidoc"));
+
+            task.setBreakingChangesTemplate(projectDirectory.file(RESOURCES + "templates/breaking-changes.asciidoc"));
+            task.setBreakingChangesFile(
+                projectDirectory.file(
+                    String.format("docs/reference/migration/migrate_%d_%d.asciidoc", version.getMajor(), version.getMinor())
+                )
+            );
+
+            task.dependsOn(validateChangelogsTask);
+        });
+
+        // FIXME
+        // project.getTasks().named("precommit").configure(task -> task.dependsOn(validateChangelogsTask));
```

Comment:
@mark-vieira I couldn't get this work after the build project restructure - any ideas?

Reply:
This is because the `precommit` task is registered later in the root `build.gradle` file. I'd apply the `PrecommitTaskPlugin` from this plugin, then change the stuff in build.gradle to configure the existing task rather than register a new one.

Reply:
@mark-vieira yep, that works.